cuML/src/glm/qn/glm_softmax.h

```diff
@@ -186,11 +188,11 @@ void launchLogsoftmax(T *loss_val, T *dldZ, const T *Z, const T *labels, int C,
 template <typename T> struct Softmax : GLMBase<T, Softmax<T>> {
   typedef GLMBase<T, Softmax<T>> Super;
 
-  Softmax(int D, int C, bool has_bias, const cublasHandle_t & cublas)
-      : Super(D, C, has_bias, cublas) {}
+  Softmax(const cumlHandle_impl &handle, int D, int C, bool has_bias)
+      : Super(handle, D, C, has_bias) {}
 
-  inline void getLossAndDZ(T *loss_val, SimpleMat<T> &Z,
-                           const SimpleVec<T> &y, cudaStream_t stream) {
+  inline void getLossAndDZ(T *loss_val, SimpleMat<T> &Z, const SimpleVec<T> &y,
+                           cudaStream_t stream) {
 
     launchLogsoftmax(loss_val, Z.data, Z.data, y.data, Z.m, Z.n, stream);
```

Comment:
launchLogsoftmax calls synchronous `cudaMemset` which due to default stream synchronization implicitly synchronizes all streams. This should be changed. Why was that missed in your tests?

Reply:
Good question.
I wrote separate test code to play around with RMM and forgot to add it.

Reply:
I ran the analysis on the unit test. Within the relevant range, there is no more activity on the default stream.